clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignor.java

```diff
@@ -132,6 +132,16 @@ public void setGroupInstanceId(Optional<String> groupInstanceId) {
         public Optional<String> groupInstanceId() {
             return groupInstanceId;
         }
+
+        @Override
+        public String toString() {
+            return "Subscription(" +
+                "topics=" + topics +
+                ", userData=" + userData +
+                ", ownedPartitions=" + ownedPartitions +
+                (groupInstanceId.isPresent() ? ", groupInstanceId=" + groupInstanceId.get() : "") +
```

Comment:
How about using lambda? `(groupInstanceId.map(s -> ", groupInstanceId=" + s).orElse(""))`

Reply:
For another, it seems to me "null" is more suitable than empty string.

Reply:
Updated. Thank you!